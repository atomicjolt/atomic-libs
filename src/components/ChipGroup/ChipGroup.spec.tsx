import React from "react";
import { describe, test, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { ChipGroup, Chip } from ".";

describe("ChipGroup", () => {
  test("matches snapshot", () => {
    const result = render(
      <ChipGroup label="My Chip Group">
        <Chip>Item</Chip>
      </ChipGroup>
    );
    expect(result.asFragment()).toMatchSnapshot();
  });

  test("renders label", () => {
    render(
      <ChipGroup label="My Chip Group">
        <Chip>Item</Chip>
      </ChipGroup>
    );
    expect(screen.getByText("My Chip Group")).not.toBeNull();
  });

  test("renders chips", () => {
    const chips = [
      { key: "1", rendered: "Chip 1" },
      { key: "2", rendered: "Chip 2" },
      { key: "3", rendered: "Chip 3" },
    ];
    render(
      <ChipGroup items={chips} label="Label">
        {({ key, rendered }) => <Chip key={key}>{rendered}</Chip>}
      </ChipGroup>
    );
    chips.forEach((chip) => {
      expect(screen.getByText(chip.rendered)).not.toBeNull();
    });
  });
});

describe("Chip", () => {
  test("matches snapshot", () => {
    const result = render(<Chip>Item</Chip>);
    expect(result.asFragment()).toMatchSnapshot();
  });
});
