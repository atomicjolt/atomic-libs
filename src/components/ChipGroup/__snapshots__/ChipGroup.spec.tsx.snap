// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fnVZQs inwVdt aj-chip"
  >
    <div
      class="sc-bqGGcD javuMF"
    >
      Item
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChipGroup > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ksltVi gsBVvp aj-chip-group is-full"
  >
    <label
      class="sc-bdnyFh hQZKgx"
      id="react-aria-:r1:"
    >
      My Chip Group
    </label>
    <div
      aria-atomic="false"
      aria-labelledby="react-aria-:r1:"
      aria-live="off"
      aria-relevant="additions"
      class="sc-hBMVwJ dncQew"
      id="react-aria-:r0:"
      role="grid"
      tabindex="0"
    >
      <div
        aria-label="Item"
        class="sc-fnVZQs inwVdt aj-chip"
        data-focus-visible="false"
        data-key="$.0"
        id="react-aria-:r0:-$.0"
        role="row"
        tabindex="0"
      >
        <div
          aria-colindex="1"
          class="sc-bqGGcD javuMF"
          role="gridcell"
        >
          Item
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
